### PR TITLE
Patch Clear Island

### PR DESCRIPTION
**Pull request by DoctaEnkoda**
The code that I submit and made in Java 16, will have to modify the functions in Java 8
https://github.com/Iridium-Development/IridiumSkyblock/issues/489

### CODE REVIEW

**Comment by PeachesMLG** [comment]
Looks like the builds failing

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java [hunk 1]
```diff
@@ -14,49 +14,49 @@
...
-     * Puts the user in the correct order of userIslandIndex
+     * Puts the user in userIslandMap
      *
      * @param user The specified User
      */
     public void resortIsland(User user) {
-        userIslandIndex.remove(user);
-        int islandIndex = Collections.binarySearch(userIslandIndex, user, Comparator.comparing(u -> u.getIsland().map(Island::getId).orElse(0)));
-        userIslandIndex.add(islandIndex < 0 ? -(islandIndex + 1) : islandIndex, user);
+        if (userIslandMap.get(user.getUuid()) != null) {
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java:44]: This method was origionally used to sort the User's via the IslandID, so we can get a list of all members in an island.

Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java:44]: But now, its not list, its linkedhashmap

**Comment by DoctaEnkoda** [comment]
He has fail because my code use function Java 11, I don't have much time to convert it to java 8.
In addition, moving to Java 11 will only do more good to continue developing on it.

**Comment by dlsf** [comment]
But we would lose 25% of our users, which is a lot: https://bstats.org/plugin/bukkit/IridiumSkyblock/5825

**Comment by DoctaEnkoda** [comment]
Build Java 8 OK

**Comment by PeachesMLG** [comment]
Have you tested this?

Im currently adding unit tests to Iridium, and im not really confident this will be bug free, ima add unit tests and then see if they all pass with this version.

**Comment by DoctaEnkoda** [comment]
With this patch, i'm 0 lost island on my server, with actual code, i'm 20 lost island / 4000 players per day

**Comment by PeachesMLG** [comment]
sorry bout the delay, ill take a look at this tmr

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java [hunk 2]
```diff
@@ -62,6 +59,30 @@ public void save() {
             exception.printStackTrace();
         }
     }
+    
+    public void saveHashMap(LinkedHashMap<?, T> uuidLinkedHashMap) {
+        this.saveHashMap(uuidLinkedHashMap, true);
+    }
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java:65]:
```suggestion
    }
    
```

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java [hunk 3]
```diff
@@ -62,6 +59,30 @@ public void save() {
...
+    public void saveHashMap(LinkedHashMap<?, T> uuidLinkedHashMap, boolean useChanged) {
+        List<T> tList = new LinkedList<>(uuidLinkedHashMap.values());
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java:67]: What's the point in the `t` prefix here?

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java [hunk 4]
```diff
@@ -62,6 +59,30 @@ public void save() {
...
+        int sizeList = tList.size();
+        try {
+            for (int i = 0; i < sizeList; i++) {
+                T t = tList.get(i);
+                if (!useChanged) { // all save
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java:72]: What exactly is the use case for this toggle?

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java (hunk 4 above)
Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java:72]: For save user data

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java (hunk 3 above)
Comment (DoctaEnkoda) [reply src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/TableManager.java:67]: Just value is T

**Comment by DoctaEnkoda** [comment]
I will include a new PR after this patch because, another error that was causing the plugin to crash was found.

**Comment by Hyronymos1337** [comment]
any update on this?

**Comment by DoctaEnkoda** [comment]
closed because does not intend to fix it

**Pull request closed without merging**